crates/bevy_xr/src/interaction.rs

```diff
@@ -0,0 +1,146 @@
+
+// Note: indices follow WebXR convention. OpenXR's palm joint is missing, but it can be retrieved
+// using `XrState::hand_motion(..., HandAction::Grip)`.
+pub const XR_HAND_JOINT_WRIST: usize = 0;
+pub const XR_HAND_JOINT_THUMB_METACARPAL: usize = 1;
+pub const XR_HAND_JOINT_THUMB_PROXIMAL: usize = 2;
+pub const XR_HAND_JOINT_THUMB_DISTAL: usize = 3;
+pub const XR_HAND_JOINT_THUMB_TIP: usize = 4;
+pub const XR_HAND_JOINT_INDEX_METACARPAL: usize = 5;
+pub const XR_HAND_JOINT_INDEX_PROXIMAL: usize = 6;
+pub const XR_HAND_JOINT_INDEX_INTERMEDIATE: usize = 7;
+pub const XR_HAND_JOINT_INDEX_DISTAL: usize = 8;
+pub const XR_HAND_JOINT_INDEX_TIP: usize = 9;
+pub const XR_HAND_JOINT_MIDDLE_METACARPAL: usize = 10;
+pub const XR_HAND_JOINT_MIDDLE_PROXIMAL: usize = 11;
+pub const XR_HAND_JOINT_MIDDLE_INTERMEDIATE: usize = 12;
+pub const XR_HAND_JOINT_MIDDLE_DISTAL: usize = 13;
+pub const XR_HAND_JOINT_MIDDLE_TIP: usize = 14;
+pub const XR_HAND_JOINT_RING_METACARPAL: usize = 15;
+pub const XR_HAND_JOINT_RING_PROXIMAL: usize = 16;
+pub const XR_HAND_JOINT_RING_INTERMEDIATE: usize = 17;
+pub const XR_HAND_JOINT_RING_DISTAL: usize = 18;
+pub const XR_HAND_JOINT_RING_TIP: usize = 19;
+pub const XR_HAND_JOINT_LITTLE_METACARPAL: usize = 20;
+pub const XR_HAND_JOINT_LITTLE_PROXIMAL: usize = 21;
+pub const XR_HAND_JOINT_LITTLE_INTERMEDIATE: usize = 22;
+pub const XR_HAND_JOINT_LITTLE_DISTAL: usize = 23;
+pub const XR_HAND_JOINT_LITTLE_TIP: usize = 24;
+pub const XR_HAND_JOINT_COUNT: usize = 25;
+
+#[derive(Clone)]
+pub struct Position {
+    pub value: Vec3,
+
+    /// If `tracked` is false, the position is inferred (for example just after a controller has
+    /// occluded)
+    pub tracked: bool,
+}
+
+#[derive(Clone)]
+pub struct Orientation {
+    pub value: Quat,
+    pub tracked: bool,
+}
+
+#[derive(Default)]
+pub struct Pose {
+    pub position: Option<Position>,
+    pub orientation: Option<Orientation>,
+}
+
+impl Pose {
+    pub fn to_mat4(&self) -> Mat4 {
+        todo!()
+    }
+}
+
+#[derive(Default)]
+pub struct Motion {
+    pub pose: Pose,
+    pub linear_velocity: Option<Vec3>,
+    pub angular_velocity: Option<Vec3>,
+}
+
+#[derive(Clone)]
+pub enum TrackingReferenceMode {
+    /// The coordinate system (position and orientation) is set as the headset pose at startup or
+    /// after a recenter. This should be used only for experiences where the user is laid down.
+    Tilted,
+
+    /// The coordinate system (position and gravity-aligned orientation) is calculated from the
+    /// headset pose at startup or after a recenter. This is for seated experiences.
+    GravityAligned,
+
+    /// The coordinate system (position and orientation) corresponds to the center of a rectangle at
+    /// floor level, with +Y up. This is for stading or room-scale experiences.
+    Stage,
+}
+
+impl Default for TrackingReferenceMode {
+    fn default() -> Self {
+        Self::GravityAligned
+    }
+}
+
+pub enum HandType {
```

Comment:
Simple enums like this should derive
#[derive(Debug, Clone, Copy, Eq, PartialEq, Hash)]`

Reply:
(as well as other simple structs such as `BinaryEvent`

Reply:
Yeah, I was not really focusing on derives for now.